osu.Game.Rulesets.Osu/Mods/OsuModSlide.cs

```diff
@@ -135,10 +132,19 @@ private void applySlideIn(DrawableHitObject drawable, ArmedState state)
                         if (counter == 0) return;
 
                         Vector2 originalPosition = originalPositions[counter - 1];
+                        Vector2 movementVector = movementVectors[counter - 1];
+
+                        if (SlideDirection.Value == SlideDirectionEnum.Random)
+                        {
+                            float length = movementVector.Length;
+                            Random random = new Random();
```

Comment:
Use `RNG` static class. Also you will want to use a set seed if you want this to be comparable across plays (and replay playback)

Reply:
your branch was outdated, this was fixed in https://github.com/ppy/osu/pull/23924/commits/db0d06768547a40cc3e1a8fb7fe68ce45a74b0c9... but i did notice a couple other bugs so i also fixed those just now. Specifically, random hitobjects all came from the same random direction (not sure how i didn't notice that), and replays had different random values (caused by applying randomness in `applySlideIn` rather than `ApplyToBeatmap`)